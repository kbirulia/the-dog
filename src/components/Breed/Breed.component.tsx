import React from "react";
import IBreed from "../../interfaces/breed";

import "./Breed.style.css";
import Property from "./Property.component";

type TBreedProps = {
  breed: IBreed;
};

const Breed = ({ breed }: TBreedProps): JSX.Element => {
  return (
    <>
      <h2 className="main__title">{breed.name}</h2>
      <ul>
        <Property name="Weight" value={`${breed.weight.metric} kg`} />
        <Property name="Height" value={`${breed.height.metric} cm`} />
        <Property name="Origin" value={breed.origin} />
        <Property name="Life Span" value={breed.life_span} />
        <Property name="Temperament" value={breed.temperament} />
        <Property name="Bred For" value={breed.bred_for} />
      </ul>
    </>
  );
};

export default Breed;
