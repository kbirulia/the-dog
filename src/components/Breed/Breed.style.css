.breed-details__item {
  display: inline-block;
  width: 135px;
  font-weight: bold;
}
