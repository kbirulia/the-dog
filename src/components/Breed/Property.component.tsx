import React from "react";

type TPropertyProps = {
  name: string;
  value: string;
};

const Property = ({ name, value }: TPropertyProps): JSX.Element => (
  <li>
    <span className="breed-details__item">{name}: </span> {value}
  </li>
);

export default Property;
