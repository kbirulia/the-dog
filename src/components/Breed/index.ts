export { default } from "./Breed.container";
