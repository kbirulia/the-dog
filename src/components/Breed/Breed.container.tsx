import React from "react";
import { useParams } from "react-router-dom";
import useFetch from "../../hooks/useFetch";
import Breed from "./Breed.component";
import FetchResults from "../FetchResults";
import IBreed from "../../interfaces/breed";

interface TParamTypes {
  breedId: string;
}

const BreedContainer = (): JSX.Element => {
  const { breedId } = useParams<TParamTypes>();

  const { error, isLoaded, data } = useFetch<IBreed | null>(
    null,
    `https://api.thedogapi.com/v1/breeds/${breedId}`
  );

  return (
    <FetchResults error={error} isLoaded={isLoaded}>
      {data && <Breed breed={data} />}
    </FetchResults>
  );
};

export default BreedContainer;
