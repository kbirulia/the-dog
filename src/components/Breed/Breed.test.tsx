import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Breed from "./Breed.component";

describe("Breed", () => {
  test("should render breed", () => {
    const tree = renderer
      .create(
        <Router>
          <Breed
            breed={{
              weight: {
                imperial: "44 - 66",
                metric: "20 - 30",
              },
              height: {
                imperial: "30",
                metric: "76",
              },
              id: 3,
              name: "African Hunting Dog",
              bred_for: "A wild pack animal",
              life_span: "11 years",
              temperament: "Wild, Hardworking, Dutiful",
              origin: "",
            }}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
