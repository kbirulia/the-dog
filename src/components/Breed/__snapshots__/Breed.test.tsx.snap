// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breed should render breed 1`] = `
Array [
  <h2
    className="main__title"
  >
    African Hunting Dog
  </h2>,
  <ul>
    <li>
      <span
        className="breed-details__item"
      >
        Weight
        : 
      </span>
       
      20 - 30 kg
    </li>
    <li>
      <span
        className="breed-details__item"
      >
        Height
        : 
      </span>
       
      76 cm
    </li>
    <li>
      <span
        className="breed-details__item"
      >
        Origin
        : 
      </span>
       
      
    </li>
    <li>
      <span
        className="breed-details__item"
      >
        Life Span
        : 
      </span>
       
      11 years
    </li>
    <li>
      <span
        className="breed-details__item"
      >
        Temperament
        : 
      </span>
       
      Wild, Hardworking, Dutiful
    </li>
    <li>
      <span
        className="breed-details__item"
      >
        Bred For
        : 
      </span>
       
      A wild pack animal
    </li>
  </ul>,
]
`;
